File: WEEK08/DAY02/8595/0.py
# from sys import stdin
# n =int(stdin.readline().strip()) # (1 ≤ n ≤ 5,000,000)
# inpt = list(stdin.readline().strip())
# ans = ''
# for hid_num in inpt:
#      ans=ans+hid_num  if hid_num.isnumeric()else ans+' '
# print(sum(list(map(int,ans.split()))))

# 시간초과 ㅜㅜ 
# https://buganddog.tistory.com/17 참고
n = int(input())
word = input()
number = [str(i) for i in range(10)]
def hidden_n():
    place_v = 1
    ans = 0 
    temp = 0 
    for i in range(n):
        i = -i -1 
        if word[i] in number:
            temp += int(word[i])*place_v
            place_v=place_v*10
            
        if i ==-n or word[i] not in number: 
            if place_v >= 10**7: #숫자가 6자리 이상이라면 종료, 
                place_v = 1
                temp = 0
                continue 
            ans += temp 
            place_v = 1
            temp = 0 
    return ans
    
print(hidden_n())

File: WEEK08/DAY02/8595/0.js
const fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString();
input = input.split("\n");
// const n = prompt("1 ≤ n ≤ 5,000,000인 단어 길이 n을 입력해주세요.");
const nums = new Array(9).fill(0);
nums.map((idx, v) => (nums[idx + 1] = idx + 1)); //[0,1,2,3,4,5,6,7,8,9]
// const word = prompt(`길이가 ${n}인 단어를 입력해주세요.`);
let [place_value, ans, temp] = [1, 0, 0];
for (let i = parseInt(n) - 1; i >= 0; i--) {
  if (nums.includes(parseInt(word[i]))) {
    temp += parseInt(word[i]) * place_value;
    place_value = place_value * 10;
  }
  if (i === 0 || !nums.includes(parseInt(word[i]))) {
    //맨앞에서부터 숫자가 시작하거나 숫자가 아닌 경우
    if (place_value >= Math.pow(10, 7)) {
      place_value = 1;
      temp = 0;
      continue;
    }
    ans += temp;
    place_value = 1;
    temp = 0;
  }
}
console.log(ans);
